### PR TITLE
Add SourceLocationGroup utility to eve

### PR DESCRIPTION
**Pull request by egparedes**
## Description

Add another utility class to eve representing a source location originated from a fusing several canonical source locations.

### CODE REVIEW

File: tests/eve_tests/unit_tests/test_type_definitions.py [hunk 1]
```diff
@@ -54,12 +54,65 @@ def test_symbol_types():
...
+        )
+        assert str(loc) == "<'dir/source.py': Line 1, Col 1 to Line 2, Col 2>"
+
+    def test_construction_from_ast(self):
+        import ast
+
+        ast_node = ast.parse("a = b + 1").body[0]
+        loc = eve.type_definitions.SourceLocation.from_AST(ast_node, "source.py")
+        print("\n", ast_node.end_lineno)
+        print(loc)
```
Comment (DropD) [inline tests/eve_tests/unit_tests/test_type_definitions.py:83]: Are these meant to be left in?

File: tests/eve_tests/unit_tests/test_type_definitions.py [hunk 2]
```diff
@@ -54,12 +54,65 @@ def test_symbol_types():
...
+
+        assert loc.line == ast_node.lineno
+        assert loc.column == ast_node.col_offset + 1
+        assert loc.source == "source.py"
```
Comment (DropD) [inline tests/eve_tests/unit_tests/test_type_definitions.py:87]: `.end_line` and `.end_column` not yet tested

**Review by DropD — COMMENTED** [review]
Looks good, just two small things

File: tests/eve_tests/unit_tests/test_type_definitions.py (hunk 1 above)
Comment (egparedes) [reply tests/eve_tests/unit_tests/test_type_definitions.py]: Not at all. Removed

File: tests/eve_tests/unit_tests/test_type_definitions.py (hunk 2 above)
Comment (egparedes) [reply tests/eve_tests/unit_tests/test_type_definitions.py:85]: Oops, tests have been updated to address this.

**Review by DropD — APPROVED** [review]

**Pull request merged**